### PR TITLE
RK-244

### PR DESCRIPTION
**Pull request by ModisAtHMRC**
added repos for ChargeReference (primary key IDs) and FraudReport; reworked controller to use new repo

### CODE REVIEW

File: app/uk/gov/hmrc/taxfraudreporting/repositories/SequentialChargeReferenceRepository.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    ) with ChargeReferenceRepository {
+
+  private val id: String = "counter"
+
+  val started: Future[Unit] =
+    Future.successful(collection.insertOne(ChargeRefJson.apply(id, 0)))
+
+  override def nextChargeReference(): Future[ChargeReference] =
+    /*
+    val selector = Json.obj("_id" -> id)
```
Comment (shinealex33) [inline app/uk/gov/hmrc/taxfraudreporting/repositories/SequentialChargeReferenceRepository.scala:44]: please remove commented codes if it is not required

File: test/uk/gov/hmrc/taxfraudreporting/integration/IntegrationSpecCommonBase.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  lazy val injector: Injector = app.injector
+
+  val sampleVRN: String               = "123456789"
+  val sampleVRN2: String              = "123456788"
+  val sampleVRN3: String              = "123456787"
+  val sampleEnrolmentKey: String      = s"HMRC-MTD-VAT~VRN~$sampleVRN"
+  val sampleEnrolmentKey2: String     = s"HMRC-MTD-VAT~VRN~$sampleVRN2"
+  val sampleEnrolmentKey3: String     = s"HMRC-MTD-VAT~VRN~$sampleVRN3"
+  val enrolmentKeyNotInMongo: String  = "HMRC-MTD-VAT~VRN~123456790"
```
Comment (shinealex33) [inline test/uk/gov/hmrc/taxfraudreporting/integration/IntegrationSpecCommonBase.scala:42]: please remove all unused variables here

**Review by shinealex33 — CHANGES_REQUESTED** [review]
Please take look on the comments.

File: app/uk/gov/hmrc/taxfraudreporting/repositories/FraudReportRepositoryImpl.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  validationService: JsonValidationService,
+  chargeReferenceService: ChargeReferenceRepository
+)(implicit val ec: ExecutionContext)
+    extends PlayMongoRepository[FraudReport](
+      collectionName = "fraud-reports",
+      mongoComponent = mongoComponent,
+      domainFormat = FraudReport.format,
+      indexes = Seq(
+        IndexModel(ascending("lastUpdated"), IndexOptions().name("fraud-reports-last-updated-index")),
+        IndexModel(ascending("state"), IndexOptions().name("fraud-reports-state-index"))
```
[the remark this answers is not in the document]
Comment (ModisAtHMRC) [reply app/uk/gov/hmrc/taxfraudreporting/repositories/FraudReportRepositoryImpl.scala]: Replaced these indices as requested.

File: app/uk/gov/hmrc/taxfraudreporting/repositories/SequentialChargeReferenceRepository.scala [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import org.mongodb.scala.model.Filters.equal
+import org.mongodb.scala.model.{FindOneAndUpdateOptions, ReturnDocument, Updates}
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+import uk.gov.hmrc.taxfraudreporting.models.{ChargeRefJson, ChargeReference}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SequentialChargeReferenceRepository @Inject() (mongoComponent: MongoComponent)(implicit ec: ExecutionContext)
```
[the remark this answers is not in the document]
Comment (ModisAtHMRC) [reply app/uk/gov/hmrc/taxfraudreporting/repositories/SequentialChargeReferenceRepository.scala]: Renamed, along with associated models and Spec

File: app/uk/gov/hmrc/taxfraudreporting/repositories/SequentialChargeReferenceRepository.scala (hunk 1 above)
Comment (ModisAtHMRC) [reply app/uk/gov/hmrc/taxfraudreporting/repositories/SequentialChargeReferenceRepository.scala]: Removed

File: test/uk/gov/hmrc/taxfraudreporting/integration/IntegrationSpecCommonBase.scala (hunk 2 above)
Comment (ModisAtHMRC) [reply test/uk/gov/hmrc/taxfraudreporting/integration/IntegrationSpecCommonBase.scala]: Removed

**Review by shinealex33 — APPROVED** [review]
LGTM

**Pull request merged**